src/main/java/teammates/logic/api/Logic.java

```diff
@@ -331,6 +331,23 @@ public void deleteInstructorCascade(String courseId, String email) {
         instructorsLogic.deleteInstructorCascade(courseId, email);
     }
 
+    /**
+     * Deletes an unregistered instructor cascade its associated demo course and students.
```

Comment:
I don't think a part of the code that checks whether the course/instructor is only demo exists.